File: app-mvp-java/src/main/java/com/example/app_mvp_java/page/main/MainActivity.java
package com.example.app_mvp_java.page.main;

import android.widget.TextView;
import android.widget.Toast;

import com.example.app_mvp_java.R;
import com.example.app_mvp_java.base.BaseActivity;

/**
 * @author wandervogel
 * @date 2025-09-22
 * @description 首页实现
 */
public class MainActivity extends BaseActivity<MainPresenter> implements MainContract.View {

    private TextView tvWelcome;

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected MainPresenter createPresenter() {
        return new MainPresenter();
    }

    @Override
    protected void initView() {
        tvWelcome = findViewById(R.id.tv_welcome);
    }

    @Override
    protected void initData() {
        mPresenter.loadData();
    }

    @Override
    public void showWelcomeMessage(String message) {
        if (tvWelcome != null) {
            tvWelcome.setText(message);
        }
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }
}

File: app-mvvm-kotlin/src/main/java/com/example/app_mvvm_kotlin/activities/MainActivity.kt
package com.example.app_mvvm_kotlin.activities


import android.widget.Toast
import com.example.app_mvvm_kotlin.R
import com.example.common.ui.activity.BaseActivity

class MainActivity : BaseActivity() {
    override fun getLayoutId() = R.layout.activity_main

    override fun initView() {
        super.initView()
        Toast.makeText(this, "测试", Toast.LENGTH_SHORT).show()
    }
}